datalad_container/containers_run.py

```diff
@@ -131,14 +132,41 @@ def __call__(cmd, container_name=None, dataset=None,
             # just prepend and pray
             cmd = container['path'] + ' ' + cmd
 
+        extra_inputs_ = container.get("extra-input",[])
+        # TODO: Can't use EnsureListOf(str) yet as it handles strings as iterables.
+        # See this PR: https://github.com/datalad/datalad/pull/7267
+        # get_all=True still returns a string for a single value
+        extra_inputs_ = [extra_inputs_] if isinstance(extra_inputs_, str) else extra_inputs_
+        extra_inputs = []
+        for extra_input in set(extra_inputs_):
+            try:
+                xi_kwargs = dict(
+                    img_dspath=image_dspath,
+                    img_dirpath=op.dirname(image_path) or ".",
+                )
+                extra_inputs.append(extra_input.format(**xi_kwargs))
+            except KeyError as exc:
+                yield get_status_dict(
+                    'run',
+                    ds=ds,
+                    status='error',
+                    message=(
+                        'Unrecognized extra_input placeholder: %s. '
+                        'See containers-add for information on known ones: %s',
+                        exc,
+                        ", ".join(xi_kwargs)))
+                return
+
+        lgr.debug("extra_inputs = {!r}".format(extra_inputs))
```

Comment:
```suggestion
        lgr.debug("extra_inputs = %r", extra_inputs)
```
no need to do formatting unless really at debug level. I know that savings are negligible but in general it is a preferred way (also to avoid surprises if `extra_inputs` has some `%` in them and then someone would add here more to the line, and some argument to interpolate and it would all break)